examples/VLCB4IN4OUT/VLCB4IN4OUT.ino

```diff
@@ -0,0 +1,550 @@
+  }
+}
+//
+///  setup Module - runs once at power on called from setup()
+//
+
+void setupModule()
+{
+  unsigned int nodeNum = modconfig.nodeNum;
+  // configure the module switches, active low
+  for (byte i = 0; i < NUM_SWITCHES; i++)
+  {
+    moduleSwitch[i].attach(SWITCH[i], INPUT_PULLUP);
+    moduleSwitch[i].interval(5);
+    switchState[i] = false;
+  }
+
+  // configure the module LEDs
+  for (byte i = 0; i < NUM_LEDS; i++) {
+    moduleLED[i].setPin(LED[i]);
+  }
+
+  Serial << "> Module has " << NUM_LEDS << " LEDs and " << NUM_SWITCHES << " switches." << endl;
+}
+
+
+void setup()
+{
+  Serial.begin(115200);
+  Serial << endl << F("> ** VLCB 4 in 4 out v1 ** ") << __FILE__ << endl;
+
+  setupVLCB();
+  setupModule();
+
+  // end of setup
+  DEBUG_PRINT(F("> ready"));
+}
+
+
+void loop()
+{
+  // do VLCB message, switch and LED processing
+  controller.process();
+
+  // process console commands
+  processSerialInput();
+
+  // Run the LED code
+  for (int i = 0; i < NUM_LEDS; i++) {
+    moduleLED[i].run();
+  }
+
+  // test for switch input
+  processSwitches();
+
+  //  processStartOfDay();
+}
+
+void processSwitches(void) {
+  bool isSuccess = true;
+  for (byte i = 0; i < NUM_SWITCHES; i++) {
+    moduleSwitch[i].update();
+    if (moduleSwitch[i].changed()) {      
+      byte nn_en[4];
+      modconfig.readEvent(i, nn_en);
+      byte evval = modconfig.getEventEVval(i, 1);
+      bool state;
+
+      DEBUG_PRINT(F("> Button ") << i << F(" state change detected"));
+      Serial << F(" EV Value = ") << evval << endl;
+
+      switch (evval) {
+        case 0:
+          // ON and OFF
+          state = (moduleSwitch[i].fell());
+          DEBUG_PRINT(F("> Button ") << i
+                      << (moduleSwitch[i].fell() ? F(" pressed, send state: ") : F(" released, send state: ")) << state);
+          epService.sendEvent(state, i);
+          break;
+
+        case 1:
+          // Only ON
+          if (moduleSwitch[i].fell()) {
+            state = true;
+            DEBUG_PRINT(F("> Button ") << i << F(" pressed, send state: ") << state);
+            epService.sendEvent(state, i);
+          }
+          break;
+
+        case 2:
+          // Only OFF
+          if (moduleSwitch[i].fell()) {
+            state = false;
+            DEBUG_PRINT(F("> Button ") << i << F(" pressed, send state: ") << state);
+            epService.sendEvent(state, i);
+          }
+          break;
+
+        case 3:
+          // Toggle button
+          if (moduleSwitch[i].fell()) {
+            switchState[i] = !switchState[i];
+            state = (switchState[i]);
+            DEBUG_PRINT(F("> Button ") << i
+                        << (moduleSwitch[i].fell() ? F(" pressed, send state: ") : F(" released, send state: ")) << state);
+            epService.sendEvent(state, i);
+          }
+
+          break;
+
+        default:
+          DEBUG_PRINT(F("> Invalid EV value."));
+          break;
+      }
+    }
+  }
+  if (!isSuccess) {
```

Comment:
Always false. isSuccess is never set to false.

Reply:
Fair point.  I'll delete it.